### PR TITLE
Adding user to user and distributor table

### PR DESCRIPTION
**Pull request by lakshmisatish-code**

### CODE REVIEW

File: src/main/java/com/revature/repositories/SupplierRepo.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.springframework.data.repository.CrudRepository;
+
+import com.revature.beans.Supplier;
+
+public interface SupplierRepo extends CrudRepository<Supplier, Integer> {
+	List<Supplier> findByName(String name);
+
+}
+
```
Comment (nickrens) [inline src/main/java/com/revature/repositories/SupplierRepo.java]: This class already exists in the master branch, why did you make another version?

File: src/main/java/com/revature/repositories/ProductRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.revature.beans.Product;
+
+public class ProductRepositoryImpl {/*implements ProductRepository {
```
Comment (nickrens) [inline src/main/java/com/revature/repositories/ProductRepositoryImpl.java]: It looks like this class serves no purpose, can you delete it?

File: src/main/java/com/revature/beans/Distributor.java [hunk 3]
```diff
@@ -26,19 +34,22 @@
...
 	@Column
 	private String phone_number;
-
+	
 	public Distributor() {
 		super();
 	}
 
-	public Distributor(int id, String name, String address, String phone_number) {
+	public Distributor(String name, String address, String phone_number) {
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/beans/Distributor.java:42]: distributor id is removed from constructor

File: src/main/java/com/revature/beans/Distributor.java [hunk 4]
```diff
@@ -26,19 +34,22 @@
...
 		super();
-		this.id = id;
+		//this.id = id;
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/beans/Distributor.java:44]: no need of assignment to ditributor_id as it is removed from constructor

File: src/main/java/com/revature/beans/Distributor.java [hunk 5]
```diff
@@ -73,8 +84,7 @@ public void setPhone_number(String phone_number) {
 
 	@Override
 	public String toString() {
-		return "Distributor [id=" + id + ", name=" + name + ", address=" + address + ", phone_number=" + phone_number
-				+ "]";
+		return "Distributor [id=" + id + ", name=" + name + ", address=" + address + ", phone_number=" + phone_number+"]";
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/beans/Distributor.java:87]: alignment changes

File: src/main/java/com/revature/beans/User.java [hunk 6]
```diff
@@ -42,15 +43,16 @@ public User() {
 	 * @param supplier_id
 	 * @param distributor_id
 	 */
-	public User(String role_name, String username, String pass,  int distributor_id) {
+	public User(String role_name, String username, String pass, int distributor_id) {
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/beans/User.java:46]: removed space

File: src/main/java/com/revature/beans/UserDistributor.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+public class UserDistributor {
+	
+	private String role_name;
+	private String username;
+	private String pass;
+
+
+	private String name;
+
+	private String address;
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/beans/UserDistributor.java:14]: New Bean class included for inserting records into user and distributor tables

File: src/main/java/com/revature/controllers/DistributorController.java [hunk 8]
```diff
@@ -12,7 +12,10 @@
 import org.springframework.web.bind.annotation.RestController;
 
 import com.revature.beans.Distributor;
+import com.revature.beans.User;
+import com.revature.beans.UserDistributor;
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/controllers/DistributorController.java:16]: I used User and Distributor beans in DistributorController file

File: src/main/java/com/revature/controllers/DistributorController.java [hunk 9]
```diff
@@ -12,7 +12,10 @@
...
 import com.revature.services.DistributorService;
+import com.revature.services.UserService;
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/controllers/DistributorController.java:18]: I used Userservice object in Distributorcontroller file

File: src/main/java/com/revature/controllers/DistributorController.java [hunk 10]
```diff
@@ -21,6 +24,9 @@
 	@Autowired
 	private DistributorService ds;
 	
+	@Autowired
+	private UserService userservice;
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/controllers/DistributorController.java:28]: object is created for UserService

File: src/main/java/com/revature/controllers/DistributorController.java [hunk 11]
```diff
@@ -41,5 +47,31 @@ public Distributor updateDistributor(@PathVariable int id, @RequestBody Distribu
...
+	@PostMapping(value = "/users/adduser", consumes = "application/json", produces = "application/json")
+	public User addUserDistributor(@RequestBody UserDistributor userdist) {
+		
+		System.out.println("From Frontend: " +userdist);
+		
+		String rolename = userdist.getRole_name();
+		String username = userdist.getUsername();
+		String userpass = userdist.getPass();
+		String distname = userdist.getName();
+		String address = userdist.getAddress();
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/controllers/DistributorController.java:59]: adding the distributor and user details in user and distributor tables

File: src/main/java/com/revature/repositories/UserRepo.java [hunk 12]
```diff
@@ -4,17 +4,13 @@
 
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
-import org.springframework.data.repository.CrudRepository;
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/repositories/UserRepo.java:7]: removed this as I am not using CrudRepository

File: src/main/java/com/revature/repositories/UserRepo.java [hunk 13]
```diff
@@ -4,17 +4,13 @@
...
+import org.springframework.data.repository.query.Param;
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/repositories/UserRepo.java:7]: this needs to be removed

File: src/main/java/com/revature/services/UserService.java [hunk 14]
```diff
@@ -28,5 +25,7 @@
 	
 	public User login(String name, String password);
 	
+	public User validateUser(String username);
+
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/services/UserService.java:29]: ValidateUser() method is to check whether the username already present in database.

File: src/main/java/com/revature/services/UserServiceImpl.java [hunk 15]
```diff
@@ -62,18 +67,13 @@ public boolean deleteUser(int id) {
 		}
 	}
 	public User login(String name, String password) {
-		//new User();
-		//String query = "SELECT u.id , u.username username, u.role_name role_name, u.distributor_id distid  from users u where u.username='"+name+"' and u.pass='" +password+ "'";
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/services/UserServiceImpl.java:66]: removed the commented lines

File: src/main/java/com/revature/services/UserServiceImpl.java [hunk 16]
```diff
@@ -62,18 +67,13 @@ public boolean deleteUser(int id) {
...
-		
+				
 		String query = "SELECT u.id, u.role_name, u.username, u.pass, u.distributor_id  from users u where u.username='"+name+"' and u.pass='" +password+ "'";
 		System.out.println("Query String: "+query);
 		  Query q = em.createNativeQuery(query, User.class);
 		  System.out.println("Resultset  : "+q.getResultList());
 		  if((q.getResultList().size()) > 0) {
 		  User user = (User) q.getResultList().get(0);
-		  //user.setId(user.getId());
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/services/UserServiceImpl.java:74]: removed the commented lines

File: src/main/java/com/revature/services/UserServiceImpl.java [hunk 17]
```diff
@@ -83,7 +83,21 @@ public User login(String name, String password) {
...
-
+	
+	public User validateUser(String username) {
+		String query = "SELECT * FROM USERS u where u.username='"+username+"'";
+		System.out.println("Query String : "+query);
+		Query q = em.createNativeQuery(query, User.class);
+		  System.out.println("Resultset  : "+q.getResultList());
+		  if((q.getResultList().size()) > 0) {
+		  User user = (User) q.getResultList().get(0);
+		  return user;
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/services/UserServiceImpl.java:94]: validateUser() method definition to check whether the username already present in database

File: src/main/java/com/revature/controllers/UserController.java [hunk 18]
```diff
@@ -72,20 +68,17 @@ public boolean deleteUser(@PathVariable("id") int id) {
...
-		}*/
+		//User user = new User();
+		System.out.println("Executing Login");
 		return userservice.login(name, password);
-		//return userservice.login(name, password);
 	}
 	
+	@PostMapping(value = "/users/validateuser/{username}", produces = "application/json")
+	public User validateUser(@PathVariable("username") String username) {
+		System.out.println("Eexecuting validateUser");
```
Comment (lakshmisatish-code) [inline src/main/java/com/revature/controllers/UserController.java:79]: endpoint for validating the username

**Review by nickrens — APPROVED** [review]

**Pull request merged**